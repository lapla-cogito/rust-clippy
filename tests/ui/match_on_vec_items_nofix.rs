#![warn(clippy::match_on_vec_items)]
#![allow(clippy::useless_vec)]
//@no-rustfix

#[clippy::msrv = "1.52.0"]
fn or_patterns() {
    let arr = vec![0, 1, 2, 3];
    match arr[1] {
        //~^ ERROR: indexing into a vector may panic
        0 | 1 => println!("0 or 1"),
        _ => println!("Hello, World!"),
    }
}
