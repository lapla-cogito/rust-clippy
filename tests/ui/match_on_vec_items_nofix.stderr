error: indexing into a vector may panic
  --> tests/ui/match_on_vec_items_nofix.rs:8:11
   |
LL |     match arr[1] {
   |           ^^^^^^ help: try: `arr.get(1)`
   |
   = note: `-D clippy::match-on-vec-items` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::match_on_vec_items)]`

error: aborting due to 1 previous error

